superset/db_engine_specs/gsheets.py

```diff
@@ -33,6 +36,12 @@
 SYNTAX_ERROR_REGEX = re.compile('SQLError: near "(?P<server_error>.*?)": syntax error')
 
 
+class GSheetsParametersType(TypedDict):
+    credentials_info: Dict[str, Any]
+    query: Dict[str, Any]
```

Comment:
gshees doesn't need query

Reply:
But it does! The user can definitely pass options to it via the query string.

Reply:
Is the query string for each individual url in the catalog?

Reply:
No, it's to pass parameters to configure the engine, eg:

```
gsheets://?list_all_sheets=1
```

This makes the dropdown show not only sheets in the catalog, but all of the user's sheets.

This can be done in the "extra" session as well, to be clear.